Classes/Service/QbankService.php

```diff
@@ -234,7 +236,19 @@ public function synchronizeMetadata(int $fileId): void
             return;
         }
 
-        $qbankRecord = $this->mediaRepository->findById($qbankId);
+        try {
+            /** @var MediaResponse $qbankRecord */
+            $qbankRecord = $this->mediaRepository->findById($qbankId);
+        } catch (RequestException $re) {
+            if (QbankUtility::qbankRequestExceptionStatesMediaIsDeleted($re)) {
+                $this->updateFileRemoteIsDeleted($fileId);
+
+                throw new \TYPO3\CMS\Core\Resource\Exception\FileDoesNotExistException(
```

Comment:
I think we need our own exception here. This exception is related to files in FAL, so it's not applicable to QBank.